__tests__/setup-go.test.ts

```diff
@@ -87,94 +99,77 @@ describe('setup-go', () => {
   afterAll(async () => {}, 100000);
 
   it('can query versions', async () => {
-    let versions: im.IGoVersion[] | null = await im.getVersions(
+    let versions: im.IGoVersion[] | null = await im.getVersionsDist(
       'https://non.existant.com/path'
     );
     expect(versions).toBeDefined();
     let l: number = versions ? versions.length : 0;
     expect(l).toBe(91);
   });
 
-  it('finds stable match for exact version', async () => {
-    os.platform = 'win32';
-    os.arch = 'x64';
-
-    // get request is already mocked
-    // spec: 1.13.7 => 1.13.7 (exact)
-    let match: im.IGoVersion | undefined = await im.findMatch('1.13.7', true);
-    expect(match).toBeDefined();
-    let version: string = match ? match.version : '';
-    expect(version).toBe('go1.13.7');
-    let fileName = match ? match.files[0].filename : '';
-    expect(fileName).toBe('go1.13.7.windows-amd64.zip');
+  it('can mock manifest versions', async () => {
+    let versions: tc.IToolRelease[] | null = await tc.getManifestFromRepo(
+      'actions',
```

Comment:
I believe we don't need the following tests in the form they currently exist:
1) [it('can mock manifest versions', async () => {](https://github.com/dmitry-shibanov/setup-go/blob/103227c6219fe2f727162d8094cbf99dcc179471/__tests__/setup-go.test.ts#L110)
2) [it('can find 1.12.17 from manifest on osx', async () => {](https://github.com/dmitry-shibanov/setup-go/blob/103227c6219fe2f727162d8094cbf99dcc179471/__tests__/setup-go.test.ts#L120)
3) [it('can find 12 from manifest on linux', async () => {](https://github.com/dmitry-shibanov/setup-go/blob/103227c6219fe2f727162d8094cbf99dcc179471/__tests__/setup-go.test.ts#L134)
4) [it('can find 10 from manifest on windows', async () => {](https://github.com/dmitry-shibanov/setup-go/blob/103227c6219fe2f727162d8094cbf99dcc179471/__tests__/setup-go.test.ts#L148)

These "it" blocks test the functions located in the actions/tool-cache repository (getManifestFromRepo, findFromManifest etc.) and we don't have tests with several functions from `installer.ts` script (getGo, installGoVersion, extractGoArchive etc.).

@maxim-lobanov